graylog2-web-interface/src/views/components/searchbar/saved-search/SavedSearchList.jsx

```diff
@@ -117,11 +127,19 @@ class SavedSearchList extends React.Component<Props, State> {
     const { selectedSavedSearch } = this.state;
     const savedSearchList = (views.list || []).map((savedSearch) => {
       return (
-        <ListGroupItem active={selectedSavedSearch === savedSearch.id}
-                       onClick={() => this.setState({ selectedSavedSearch: savedSearch.id })}
-                       key={savedSearch.id}>
-          {savedSearch.title}
-        </ListGroupItem>
+        <ViewLoaderContext.Consumer key={savedSearch.id}>
+          {(loaderFunc) => (
+            <ListGroupItem active={selectedSavedSearch === savedSearch.id}
+                           onClick={() => this.onLoad(savedSearch.id, loaderFunc)}>
+              {savedSearch.title}
+              <span>
+                <DeleteButton bsSize="xsmall" bsStyle="danger" onClick={(e) => this.onDelete(e, savedSearch.id)}>
+                  <Icon name="trash" ttile="Delete" data-testid={`delete-${savedSearch.id}`} />
+                </DeleteButton>
```

Comment:
The `DeleteButton` can't be accessed with the keyboard right now. Maybe we can use the `IconButton` in the future instead. https://github.com/Graylog2/graylog2-server/pull/8064

Reply:
When the future is ready, so am I ;)